examples/ssao/main.js

```diff
@@ -0,0 +1,701 @@
+                .onChange( this.updateRadius.bind( this ) );
+            ssaoFolder.add( this._config, 'bias', biasBounds[ 0 ], biasBounds[ 1 ] )
+                .onChange( this.updateBias.bind( this ) );
+            ssaoFolder.add( this._config, 'intensity', intensityBounds[ 0 ], intensityBounds[ 1 ] )
+                .onChange( this.updateIntensity.bind( this ) );
+
+            blurFolder.add( this._config, 'crispness', 0.1, 2.5 )
+                .onChange( this.updateFloatData.bind( this, this._blurUniforms.uCrispness ) );
+
+        },
+
+        bindProjectionUpdateCallback: function ( camera ) {
+
+            var self = this;
+            var rootCam = self._viewer.getCamera();
+
+            osg.mat4.copy( camera.getProjectionMatrix(), rootCam.getProjectionMatrix() );
+
+            camera.setClampProjectionMatrixCallback( function ( m, computedNear, computedFar, nearFarRatio ) {
+
+                var viewport = rootCam.getViewport();
+                var width = viewport.width();
+                var height = viewport.height();
+
+                if ( self._previousViewport[ 0 ] !== width || self._previousViewport[ 1 ] !== height )
+                    self.resizeEvent();
+
+                this.clampProjectionMatrix( m, computedNear, computedFar, nearFarRatio );
+
+                osg.mat4.copy( camera.getViewMatrix(), rootCam.getViewMatrix() );
+                osg.mat4.copy( camera.getProjectionMatrix(), m );
+
+                var frustum = {};
+                osg.mat4.getFrustum( frustum, m );
+
+                self._aoUniforms.uNear.setFloat( frustum.zNear );
+                self._aoUniforms.uFar.setFloat( frustum.zFar );
+
+                // Projection scale used to project
+                // the SSAO radius into screen space
+                var vFov = m[ 15 ] === 1 ? 1.0 : 2.0 / m[ 5 ];
+                var scale = -2.0 * Math.tan( vFov * 0.5 );
+                var projScale = height / scale;
+                self._aoUniforms.uProjScale.setFloat( projScale );
+
+                self._projectionInfo[ 0 ] = -2.0 / ( width * m[ 0 ] );
+                self._projectionInfo[ 1 ] = -2.0 / ( height * m[ 5 ] );
+                self._projectionInfo[ 2 ] = ( 1.0 - m[ 8 ] ) / m[ 0 ];
+                self._projectionInfo[ 3 ] = ( 1.0 - m[ 9 ] ) / m[ 5 ];
+
+                self._aoUniforms.uProjectionInfo.setFloat4( self._projectionInfo );
+
+                self._previousViewport[ 0 ] = width;
+                self._previousViewport[ 1 ] = height;
+
+            } );
+
+        },
+
+        run: function () {
+
+            var self = this;
+
+            this.initDatGUI();
+            this.createViewer();
+            self.createScene();
+
+            this.readShaders().then( function () {
+
+                // Loads the glTF Sponza scene by default if exists
+                loadFiles( self, '../media/models/sponza.gltf', 'sponza' );
```

Comment:
It would be better to use the same system like for PBR, drag and drop some predefined url